collect_app/src/main/java/org/odk/collect/android/widgets/CounterWidget.kt

```diff
@@ -0,0 +1,85 @@
+package org.odk.collect.android.widgets
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import androidx.core.widget.doOnTextChanged
+import org.javarosa.core.model.data.IAnswerData
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.databinding.CounterWidgetBinding
+import org.odk.collect.android.formentry.questions.QuestionDetails
+import org.odk.collect.android.widgets.utilities.StringWidgetUtils
+
+class CounterWidget(
+    context: Context,
+    questionDetails: QuestionDetails
+) : QuestionWidget(context, questionDetails) {
+    lateinit var binding: CounterWidgetBinding
+
+    init {
+        render()
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onCreateAnswerView(
+        context: Context,
+        prompt: FormEntryPrompt,
+        answerFontSize: Int
+    ): View {
+        binding = CounterWidgetBinding.inflate(LayoutInflater.from(context))
+
+        binding.value.doOnTextChanged { text, _, _, _ ->
+            if (text.isNullOrBlank() || Integer.parseInt(text.toString()) == 0) {
```

Comment:
Did you consider using an `Int` property to track that values and a custom setter that handles updating the text and buttons for this? I'm wondering if that would end up being a bit cleaner.

Reply:
I didn't when I crated this pr but now I agree it makes sense.